creator/ingest_runs/data_generator/study_generator.py

```diff
@@ -0,0 +1,572 @@
+            if filepath.endswith(HARM_GF_EXT):
+                dt = "gVCF"
+            else:
+                dt = None
+            return dt
+
+        def workflow_type(data_type):
+            """ Return genomic workflow type based on data type """
+            if data_type in {"gVCF"}:
+                wt = "alignment"
+            else:
+                wt = None
+            return wt
+
+        # Merge all dfs
+        bio_df = self.dataframes["bio_manifest.tsv"]
+        gf_df = self.dataframes["sequencing_manifest.tsv"]
+        df = pandas_utils.merge_wo_duplicates(bio_df, gf_df, on="sample_id")
+        # Add additional columns
+        df["Data Type"] = df["harmonized_path"].map(lambda fp: data_type(fp))
+        df["Workflow Type"] = df["Data Type"].map(lambda dt: workflow_type(dt))
+        df["Cavatica ID"] = create_hex(25)
+        df["Cavatica Task ID"] = (
+            "-".join([create_hex(i) for i in [8, 4, 4, 4, 12]])
+        )
+        # Rename columns
+        col_map = {
+            "kf_id_participant": "KF Participant ID",
+            "kf_id_biospecimen": "KF Biospecimen ID",
+            "kf_id_family": "KF Family ID",
+            "harmonized_path": "Filepath",
+            "source_path": "Source Read",
+        }
+        df = df[
+            list(col_map.keys()) +
+            ["Data Type", "Workflow Type", "Cavatica ID", "Cavatica Task ID"]
+        ].rename(columns=col_map)
+
+        return df
+
+    def _assert_load(self):
+        """
+        Ensure all kf ids in the data files got loaded into Data Service
+        """
+        # Extract kf ids
+        bio_df = self.dataframes["bio_manifest.tsv"]
+        gf_df = self.dataframes["sequencing_manifest.tsv"]
+        kfids = {
+            "families": bio_df[["kf_id_family"]],
+            "participants": bio_df[["kf_id_participant"]],
+            "biospecimens": bio_df[["kf_id_biospecimen"]],
+            "genomic-files": gf_df[["kf_id_genomic_file"]],
+        }
+        for endpoint, df in kfids.items():
+            self.logger.info(f"Check all {endpoint} were loaded")
+            # Check kf ids
+            kfids = df[df.columns[0]].drop_duplicates().values.tolist()
+            base_url = f"{self.dataservice_url}/{endpoint}"
+            for kfid in kfids:
+                url = f"{base_url}/{kfid}"
+                self.logger.info(f"Checking {url}")
+                resp = requests.get(url)
+                resp.raise_for_status()
+
+            # Check exact count
+            params = {"study_id": self.study_id}
+            count = len(kfids)
+            if endpoint == "genomic-files":
+                params["harmonized"] = False
+            resp = requests.get(base_url, params=params)
+            total = resp.json()["total"]
+            assert total == count, (
+                f"{endpoint} expected count {count} != {total} found"
+            )
+
+
+def delete_entities(dataservice_url, study_id=None, logger=None):
```

Comment:
This feature will be added to kf-utils-python once https://github.com/kids-first/kf-utils-python/pull/24 merges. We should remove this function from here and import it from kf-utils-python once that PR merges.